compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentKind.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.projects;
+
+/**
+ * Represents the kind of the {@code Document} instance.
+ * <p>
+ * There are two known kinds of documents: USER_PROVIDED and GENERATED.
+ *
```

Comment:
Do we need to make this a public API for now?

Reply:
No not required to be exposed. We can keep it private or not include it at all

Reply:
Removed the DocumentKind in ab4ba20